### DISCUSSION TITLE
Complex routes with multiply variables

### DISCUSSION THREAD

**Discussion by mikeyakymenko** [vercel/next.js#40202] [Help] [answered]
👋 there! Guys, I have a question. I have project Python/SSR and I wanna rebuild it with next.js. So, this project have a realy complex URL's

like example: `/season/$var/league/$var/games/` or `/season/$var/league/$var/overall/`. But, how create urls (pages) like this with next.js, this is actually possible?

**Comment by alex289** [comment]
Hey, I think you are looking for something like catch all routes.

https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes

**Comment by adxmcollins** [comment]
Catch all routes might be the way to go with this one, but you can also use the pages folder structure to achieve this:

![Next.js pages directory structure](https://user-images.githubusercontent.com/25459663/188319388-60727958-c6da-49fd-9da2-0737dade5662.png)

It coule be easier to validate than using catch all routes, especially if you know most of the URL structure and it remains the same.

**Comment by mikeyakymenko** [comment, marked as answer]
Actually, old question, but fixed completely with appDir in next 13 ^^
